app/js/trouble.js

```diff
@@ -53,15 +54,18 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
+				for (var i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			
+			setTimeout(function() { setDropdownHeight(); }, 1000);

```

Comment:
While this does remove the "making" of the function in the loop, what's desired here is to either declare the function outside the loop and pass in the name of the function as the first argument, or to assign the anonymous function to a variable and pass that in. By invoking the function here, it means the function will be called as soon as the code executes rather than once the timeout even fires.